lib/auth.spec.ts

```diff
@@ -0,0 +1,90 @@
+import Bluebird from 'bluebird';
+import nock from 'nock';
+import { getSdk, JellyfishSDK } from './index';
+
+let context: {
+	sdk: JellyfishSDK;
+	session: string;
+	token: string;
+	executeThenWait: (
+		asyncFn: any,
+		waitQuery: any,
+		times?: number,
+	) => Promise<any>;
+};
+
+const API_URL = 'https://test.ly.fish';
+
+beforeAll(async () => {
+	context = {
+		sdk: getSdk({
+			apiPrefix: 'api/v2',
+			apiUrl: API_URL,
+		}),
+		session: 'foobar-session',
+		token: 'foobar-token',
+		executeThenWait: async (asyncFn, waitQuery, times = 20) => {
+			if (times === 0) {
+				throw new Error('The wait query did not resolve');
+			}
+
+			if (asyncFn) {
+				await asyncFn();
+			}
+
+			const results = await context.sdk.query(waitQuery);
+			if (results.length > 0) {
+				return results[0];
+			}
+
+			await Bluebird.delay(1000);
+			return context.executeThenWait(null, waitQuery, times - 1);
+		},
+	};
+});
+
+beforeEach(() => {
+	nock.cleanAll();
+});
+
+afterEach(() => {
+	context.sdk.cancelAllStreams();
+	context.sdk.cancelAllRequests();
+});
+
+test('.login() should set both token and session ID correctly if they are present in the response', async () => {
+	const { sdk } = context;
+
+	const server = nock(API_URL);
+
+	const mockSessionId = context.session;
+	const mockAuthToken = context.token;
+
+	server.post(/action/).reply((_uri) => {
```

Comment:
I think that this will probably need to be changed as we recently switched to [dedicated endpoints for login](https://github.com/product-os/jellyfish-client-sdk/pull/528/files). Can you rebase this PR and fix anything that needs changes?

Reply:
Yes, on it.

Reply:
@LucianBuzzo the changes were applied, and the PR rebased. Please let me know if there is anything else you see that needs to be done.